storage/badger/store.go

```
@@ -21,43 +21,236 @@ package badger
+	ioutil.WriteFile(lockPath, []byte("emulatorLock"), 0755)
+}
+
+func (s *Store) unlockGit() {
+	lockPath := fmt.Sprintf("%s/.git/index.lock", s.path)
+	os.Remove(lockPath)
+}
+
+func (s *Store) JumpToContext(context string) error {
+	s.unlockGit()
+	defer s.lockGit()
+	err := s.db.Close()
+	if err != nil {
+		return err
+	}
+
+	s.newCommit(fmt.Sprintf("Context switching to: %s", context))
+
+	w, _ := s.dbGitRepository.Worktree()
+
+	branch := fmt.Sprintf("refs/heads/%s", context)
+	b := plumbing.ReferenceName(branch)
+
+	// checkout branch ( first branch name is actualy context name )
+	err = w.Checkout(&git.CheckoutOptions{Create: false, Force: true, Branch: b})
+
+	if err != nil {
+		// branch doesn't exist, it means we need to create it ( first branch is named after context )
+		err := w.Checkout(&git.CheckoutOptions{Create: true, Force: true, Branch: b})
+		if err != nil {
+			return err
+		}
+
+		//after we create a tag pointing to start of this context
+		created, err := setTag(s.dbGitRepository, context, defaultSignature("Emulator", "emulator@onflow.org"))
+		if err != nil && !created {
+			return err
+		}
+
+	} else {
+
+		//create new branch
+		uuidWithHyphen := uuid.New()
+		uuid := strings.Replace(uuidWithHyphen.String(), "-", "", -1)
+
+		err := w.Checkout(&git.CheckoutOptions{Create: true, Force: true, Branch: plumbing.NewBranchReferenceName(uuid)})
+		if err != nil {
+			return err
+		}
+
+		//we have new branch but we don't need to create a tag, we just need to reset to tag
+		tag := getTag(s.dbGitRepository, context)
+		if tag != nil && !tag.Hash.IsZero() {
+			commit, _ := tag.Commit()
+			w.Reset(&git.ResetOptions{
+				Mode:   git.HardReset,
+				Commit: commit.Hash,
+			})
+		}
+
+	}
+
+	s.db, err = badger.Open(s.badgerOptions)
+	if err != nil {
+		return fmt.Errorf("could not open database: %w", err)
+	}
+
+	return nil
+
+}
+
+func (s *Store) newCommit(message string) {
+	s.unlockGit()
+	defer s.lockGit()
+	s.Sync()
```

Explicity ignore error if you are sure there is no need to handle it